contracts/SolutionSubmitter.sol

```diff
@@ -0,0 +1,73 @@
+pragma solidity ^0.5.0;
+
+import "./BatchExchange.sol";
+import "@openzeppelin/contracts/ownership/Ownable.sol";
+
+interface GasToken {
+    function freeUpTo(uint256 value) external returns (uint256 freed);
+}
+
+/** @title SolutionSubmitter - A simple wrapper contract allowing to submit settlements
+ * to the BatchExchange contract while saving gas via the GST2 or CHI GasToken
+ */
+contract SolutionSubmitter is Ownable {
+    BatchExchange public exchange;
+    GasToken public gasToken;
+    uint256 public gasPriceThreshold;
+
+    constructor(
+        BatchExchange _exchange,
+        GasToken _gasToken,
+        uint256 _gasPriceThreshold
+    ) public {
+        exchange = _exchange;
+        setGasToken(_gasToken);
+        setGasPriceThreshold(_gasPriceThreshold);
+    }
+
+    function setGasToken(GasToken _gasToken) public onlyOwner {
+        gasToken = _gasToken;
+    }
+
+    function setGasPriceThreshold(uint256 _gasPriceThreshold) public onlyOwner {
```

Comment:
ubernit: According to the solidity style guide https://solidity.readthedocs.io/en/v0.7.4/style-guide.html#avoiding-naming-collisions
```suggestion
    function setGasPriceThreshold(uint256 gasPriceThreshold_) public onlyOwner {
```

That being said, it does feel a little unnatural to me to use trailing underscores, but its standard in Solidity so maybe we should prefer it?

Reply:
I would be fine to leave it here as the trailing underscore is unlikely to even be seen... Unless that is the intention of the style guide.